core/src/lib/tree.h

```diff
@@ -120,17 +131,17 @@ struct s_tree_root {
    *  do initialization of child */
   rblink sibling{};
   rblist child;
-  const char* fname{};          /* file name */
-  int32_t FileIndex{};          /* file index */
-  uint32_t JobId{};             /* JobId */
-  int32_t delta_seq{};          /* current delta sequence */
-  uint16_t fname_len{};         /* filename length */
-  unsigned int type : 8;        /* node type */
-  unsigned int extract : 1;     /* extract item */
-  unsigned int extract_dir : 1; /* extract dir entry only */
-  unsigned int have_link : 1;   /* set if have hard link */
-  unsigned int inserted : 1;    /* set when newly inserted */
-  unsigned int loaded : 1;      /* set when the dir is in the tree */
+  const char* fname{};                      /* file name */
+  int32_t FileIndex{};                      /* file index */
+  uint32_t JobId{};                         /* JobId */
+  int32_t delta_seq{};                      /* current delta sequence */
+  uint16_t fname_len{};                     /* filename length */
+  TreeNodeType type{TreeNodeType::UNKNOWN}; /* node type */
+  unsigned int extract : 1;                 /* extract item */
```

Comment:
As far as I've looked up:
* in terms of size, using an enum instead of the int will not have much of an impact
* in terms of speed, I tried the tree building and marking in our benchmark tests (with millions of files) and there were no difference

Reply:
https://godbolt.org/z/a7s7orjK3
Its an extra ~10% memory per node (around 1.6MB per million files).
This is not nothing, though i do agree its not that bad.
I guess this is fine.